File: exercises/instructor-solutions/common-example/spring/src/main/java/com/example/restsimple/repository/StudentRepository.java
package com.example.restsimple.repository;

import com.example.restsimple.domain.Student;
import java.util.List;
import java.util.Optional;

public interface StudentRepository {
    List<Student> findAll();

    Optional<Student> findById(long id);

    boolean existsByEmail(String email);

    boolean existsByStudentNumber(String studentNumber);

    Student save(Student student);
}

File: kotlin/rest-simple-exercise/src/main/kotlin/com/example/restsimple/repository/StudentRepository.kt
package com.example.restsimple.repository

import com.example.restsimple.model.Student
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import java.util.*

interface StudentRepository : JpaRepository<Student, String> {
    // TODO implement findById, deleteById, updateById
}
